srcs/runners_manager/vm_creation/gcloud/GcloudManager.py

```diff
@@ -70,6 +71,9 @@ def update_vm_metadata(self, instance_name: str, metadata: dict):
 
             labels = instance.labels or {}
             for key, value in metadata.items():
+                value = value[:63]
+                value = value.lower()
+                value = ''.join(c if c.islower() or c.isdigit() or c in ['_', '-'] else '-' for c in value)
```

Comment:
I have a better time understanding the code with a regex:
```suggestion
                value = re.sub(r'\W', '-', value)
```

Not to forget the proper import:
```python
import re
```

Reply:
Wouldn't this be better? That way, we also keep the `_` when they are present. In reality, it's not that important:
`value = re.sub(r'[^a-z0-9_-]', '_', value)`

Reply:
Done